tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AdoNet/DbScopeFactory.cs

```diff
@@ -50,6 +53,14 @@ internal static class DbScopeFactory
                     return null;
                 }
 
+                if (Iast.Iast.Instance.Settings.Enabled)
+                {
+                    IastModule.OnSqlQuery(commandText, integrationId);
+                }
+
+                // We might block the SQL call from RASP depending on the query
+                RaspModule.OnSqlI(commandText, integrationId);
```

Comment:
As this could throw a blocking exception don't we want to do this after both try / catch blocks to ensure everything else happens?

Reply:
If we block the exception, we don't want the tracer to report a request that actually was never executed, but we want IAST to report the vulnerability.

Reply:
That's an interesting point 🤔 What if there's no parent span? Where does it get reported? because it attaches to span data right, does it generate it's own span in that case?

Reply:
Both IAST and RASP try to get the root span and return without doing anything if it's not available.